Support of versions of cp2k below 7 marked as deprecated

mtaillefumier opened the pull request:
Update of the sirius package forces cleaning up the recipe for cp2k.

dev-zero commented:
Not really a fan of this, tbh. I was always under the impression that one of the goals of Spack is to maintain access to older versions of a software package.

And, the wannier90 interface goes via files and there is no CP2K preprocessor flag for that, so it is always built-in.

toxa81 commented:
@dev-zero access to old versions is available with tagged releases of spack

mtaillefumier commented:
Wannier90 : Indeed there is no flags for wannier90. I will remove it

dev-zero commented:
also, as per policy, those versions first have to marked with `deprecated=True` and can then be removed after the next Spack release.

Commit:
Mark versions of cp2k below 9.1 as deprecated

akamishra2826 commented:
With AMD-ZSS-toolchains we are supporting cp2k-7.1 to build and run using spack, why this PR is required please explain in detail ?

mtaillefumier commented:
I reverted everything and put versions below 9.1 as deprecated

mtaillefumier commented:
The reason is that one dependency is been cleaned up and that early version of cp2k are still depending on it. I reverted all my changes and just put all versions below 9.1 as deprecated. 

Any reasons why the AMD-ZSS-toolchains need 7.1 ? (NB : cp2k-7.1 is not suppressed, just deprecated)

ningli-amd commented:
As CP2K uses a lot of modern Fortran features, support to the newer versions by LLVM-based compilers can be problematic. For example, I experienced a number of compile-time and run-time issues with v2023.1. v7.1 can be reliably supported by these compilers. I suggest we keep v7.1 onwards in the Spack recipe until the compilers catch up.

Commit:
Updated list of deprecated versions. Anything below 7.1 (excluded) is marked as deprecated

Commit:
Make pretty

Commit:
Mark SIRIUS <=6 as conflict

SIRIUS was introduced in version 7 of cp2k but could be used
in practice in version 9 (input format and functionalities).
SIRIUS with version 6 and below are marked as a dependency
conflict until CP2K version 9.

mtaillefumier commented:
I think I found a compromise. SIRIUS version below 6 are removed from the SIRIUS package (we do not support them anymore) and initial SIRIUS support in CP2K was introduced around version 7 ~ 8 but effectively usable starting from CP2K 9.x. 
I do not think SIRIUS is an issue for compiling CP2K on LLVM based FORTRAN compilers as SIRIUS is pure C++. To be able to update the SIRIUS package and keep changes to CP2K package minimal, SIRIUS 6 dependency is marked as conflict.

amd-toolchain-support reviewed (approved):
This looks fine from our POV. 

In general we would definitely prefer keeping support for packages in Spack for as long as possible. There are definitely cases where running older software versions on the latest architectures is useful and desirable, but we accept that has to be balanced with practicality of maintaining recipes if the complexity starts to grow.

msimberg commented on var/spack/repos/builtin/packages/cp2k/package.py:
```diff
@@ -231,14 +247,13 @@ class Cp2k(MakefilePackage, CudaPackage, CMakePackage, ROCmPackage):
-        # sirius support was introduced in 7+
-        conflicts("@:6")
+        # sirius support was introduced in 7, but effectively usable starting from CP2K 9
+        conflicts("@:7")
```
```suggestion
        conflicts("@:8")
```
? If the intention is to conflict with versions older than 9.

mtaillefumier committed:
Update var/spack/repos/builtin/packages/cp2k/package.py

Co-authored-by: Mikael Simberg <mikael.simberg@iki.fi>

msimberg commented:
@dev-zero would you mind having another look?

dev-zero reviewed (approved):
ACK, thanks for updating the PR and taking care of this.
